src/server/pfs/server/api_server_new_storage.go

```diff
@@ -61,3 +62,53 @@ func (a *apiServer) GetTar(request *pfs.GetTarRequest, server pfs.API_GetTarServ
 	glob := request.File.Path
 	return a.driver.getFilesNewStorageLayer(server.Context(), repo, commit, glob, grpcutil.NewStreamingBytesWriter(server))
 }
+
+func (a *apiServer) GetTarConditional(server pfs.API_GetTarConditionalServer) (retErr error) {
+	if !a.env.NewStorageLayer {
+		return fmt.Errorf("new storage layer disabled")
+	}
+	request, err := server.Recv()
+	if err != nil {
+		return err
+	}
+	func() { a.Log(request, nil, nil, 0) }()
+	defer func(start time.Time) { a.Log(request, nil, retErr, time.Since(start)) }(time.Now())
```

Comment:
I think it might actually be good to put this line above the initial `Recv()` that way if we encounter an error from that call at least we'll get a log line about it.

Reply:
How about I put both log lines immediately after the `Recv()` but before the error check? This will allow us to have the request for the initial log if it did not error. Also, I think I am going to move the feature flag check to after the log lines as well.

Reply:
:+1: that sounds good to me.

Reply:
Was this done?

Reply:
Yeah, it is in the PR now.